pixeltable/catalog/table.py

```diff
@@ -1009,3 +1012,14 @@ def __dir__(self) -> list[str]:
 
     def _ipython_key_completions_(self) -> list[str]:
         return list(self._schema.keys()) + self._query_names
+
+    class QueryScope:
```

Comment:
Since you already need to introduce this, why not have it record the QueryTemplateFunctions itself (so you only need one query-related member var in Table).

Also, please move this to the top, that makes reading the file top-to-bottom easier.